apps/frontend-pwa/src/components/groupActivity/GroupActivityListSubscriber.tsx

```diff
@@ -1,52 +1,80 @@
 import {
   GroupActivity,
   GroupActivityEndedDocument,
-  ParticipantLearningData,
+  GroupActivityStartedDocument,
 } from '@klicker-uzh/graphql/dist/ops'
 import { Dispatch, SetStateAction, useEffect } from 'react'
 
 interface GroupActivityListSubscriberProps {
   courseId: string
   subscribeToMore: any
   setEndedGroupActivity: Dispatch<SetStateAction<string | undefined>>
+  setStartedGroupActivity: Dispatch<SetStateAction<string | undefined>>
 }
 
 function GroupActivityListSubscriber({
   courseId,
   subscribeToMore,
   setEndedGroupActivity,
+  setStartedGroupActivity,
 }: GroupActivityListSubscriberProps) {
   useEffect(() => {
     subscribeToMore({
       document: GroupActivityEndedDocument,
       variables: { courseId },
       updateQuery: (
-        prev: ParticipantLearningData,
+        prev: { groupActivities: GroupActivity[] },
         {
           subscriptionData,
         }: {
           subscriptionData: { data: { groupActivityEnded: GroupActivity } }
         }
-      ) => {
+      ): { groupActivities: GroupActivity[] } => {
         if (!subscriptionData.data) return prev
 
         // trigger toast for ended group activity
-        setEndedGroupActivity(subscriptionData.data.groupActivityEnded.id)
+        const updatedActivity = subscriptionData.data.groupActivityEnded
+        setEndedGroupActivity(updatedActivity.displayName)
 
-        // update the values returned by the course overview data query
-        return {
-          ...prev,
-          course: {
-            ...prev.course,
-            groupActivities: prev.course?.groupActivities?.map((activity) =>
-              activity.id === subscriptionData.data.groupActivityEnded.id
-                ? {
-                    ...subscriptionData.data.groupActivityEnded,
-                  }
-                : activity
-            ),
-          },
+        // update the values returned by the course group activity data query
+        const updatedQueryContent = prev.groupActivities.map((activity) =>
+          activity.id === updatedActivity.id ? updatedActivity : activity
+        )
+
+        return { groupActivities: updatedQueryContent }
+      },
+    })
+
+    subscribeToMore({
+      document: GroupActivityStartedDocument,
+      variables: { courseId },
+      updateQuery: (
+        prev: { groupActivities: GroupActivity[] },
+        {
+          subscriptionData,
+        }: {
+          subscriptionData: { data: { groupActivityStarted: GroupActivity } }
+        }
+      ): { groupActivities: GroupActivity[] } => {
+        if (!subscriptionData.data) return prev
+
+        // required saveguard since the subscription is somehow triggered twice
```

Comment:
@rschlaefli Do you have an idea why this submission is triggered twice? The useEffect and the backend mutation are only executed once, the subscription in the backend is only triggered once, however, the subscription here is executed twice...